apache/trafficcontrol
# GitHub Actions workflow to validate go.mod, go.sum, and vendor directory

## What does this PR (Pull Request) do?


As a follow-up to #5229, this PR
- Adds a GitHub Actions workflow to verify that `go.mod`, `go.sum`, and the `vendor` directory have no uncommitted Git changes when modules are vendored.
- Removes an unused dependency and updates `go.sum`
- Gets `go mod vendor` to run successfully in the CiaB Traffic Ops `integration` service's Dockerfile by including the `v4-client` directory. Error from `go mod vendor` that is fixed:
  ```go
  #29 28.09 github.com/apache/trafficcontrol/traffic_ops/testing/api/v4 tested by
  #29 28.09       github.com/apache/trafficcontrol/traffic_ops/testing/api/v4.test imports
  #29 28.09       github.com/apache/trafficcontrol/traffic_ops/v4-client: no matching versions for query "latest"
  ```
- Gets the Dockerized TO and TM unit tests to working again by including go.mod/go.sum in the Docker image.
- [x] is not related to any Issue

## Which Traffic Control components are affected by this PR?


- CDN in a Box
- CI tests

## What is the best way to verify this PR?

* Verify the GitHub Action is working as intended.
  - Check the result of the *Check Go modules* workflow at 0e54b997ff. The following steps are expected to have failed:
    * Untracked changes to vendored dependencies
    * Changes to go.sum

  - Check the result of the *Check Go modules* workflow at fb9f772b25. It is expected to succeed.

* Run the TO API tests in CDN in a Box. On my computer, it took 55 minutes from the time when the API tests started running.
  ```shell
  docker-compose -f docker-compose.yml -f docker-compose.traffic-ops-test.yml up --no-deps -d edge dns db smtp trafficops-perl trafficops trafficvault integration;
  docker-compose -f docker-compose.traffic-ops-test.yml logs -f integration;
  ```

* Run the Traffic Monitor unit tests in Docker
  ```shell
  docker-compose -f traffic_monitor/tests/docker-compose.yml up --build
  ```

* Run the Traffic Ops unit tests in Docker
  ```shell
  docker-compose -f traffic_ops/app/bin/tests/docker-compose.yml up --build
  ```

## If this is a bug fix, what versions of Traffic Control are affected?

- master (09e25013fd)

## The following criteria are ALL met by this PR
